api/models/workflow.py

```diff
@@ -225,8 +225,9 @@ def tool_published(self) -> bool:
         from models.tools import WorkflowToolProvider
 
         return (
-            db.session.query(WorkflowToolProvider).filter(WorkflowToolProvider.app_id == self.app_id).first()
-            is not None
+            db.session.query(WorkflowToolProvider)
+            .filter(WorkflowToolProvider.tenant_id == self.tenant_id, WorkflowToolProvider.app_id == self.app_id)
```

Comment:
1. use the index defined in WorkflowToolProvider, see
db.UniqueConstraint("tenant_id", "app_id", name="unique_workflow_tool_provider_app_id")
2. use count() instead of first() for another better performance